docs/quickstarts/javascript/react/conditional-ui-rendering.mdx

```diff
@@ -0,0 +1,290 @@
+
+```javascript
+{
+  "roles": {
+    "viewer": {
+      "description": "A viewer with no access to sensitive asset",
+      "perms": {
+        "asertodemo.GET.api.protected": {
+          "allowed": false,
+          "enabled": false,
+          "visible": false
+        }
+      }
+    },
+    "editor": {
+      "description": "A editor with access to sensitive asset",
+      "perms": {
+        "asertodemo.GET.api.protected": {
+          "allowed": false,
+          "enabled": true,
+          "visible": false
+        }
+      }
+    },
+    "admin": {
+      "description": "An admin user with access to sensitive asset",
+      "perms": {
+        "asertodemo.GET.api.protected": {
+          "allowed": true,
+          "enabled": true,
+          "visible": true
+        }
+      }
+    }
+  }
+}
+```
+
+## Push the policy
+
+Commit, tag and push the changes you made:
+
+```
+git commit -m "updated policy to include visible and enabled clauses"
+git tag v0.0.3
+git push origin master
+git push --tags
+```
+
+## Use The Display State Map middleware
+
+Aserto provides a utility called Display State Map that will return the `visible` and `enabled` decisions from the authorizer to the server and finally to the application. To enable support for the Display State Map, we first need to update the Express.js service: let's add the reference to `displayStateMap` in the object destructuring for the `express-jwt-aserto` dependencies:
+
+Change the line:
+
+```javascript
+const { jwtAuthz } = require("express-jwt-aserto");
+```
+
+to:
+
+```javascript
+const { displayStateMap, jwtAuthz } = require("express-jwt-aserto");
+```
+
+After the `app.use(cors())` statement, add the following line:
+
+```javascript
+app.use(displayStateMap(authzOptions));
+```
+
+## Update the Application
+
+Next, we’ll add the Aserto React SDK to our application:
+
+```
+yarn add @aserto/aserto-react
+```
+
+In `index.js` add the following dependency after the `oidc-react` dependency:
+
+```javascript
+import { AsertoProvider } from "@aserto/aserto-react";
+```
+
+We'll wrap the `<App />` component with the `<AsertoProvider>` provider:
+
+```javascript
+<React.StrictMode>
+  <AuthProvider {...configuration}>
+    <AsertoProvider>
+      <App />
+    </AsertoProvider>
+  </AuthProvider>
+</React.StrictMode>
+```
+
+In `App.js`, we’ll add the `userAserto` hook, as well as a stateful object that will track the result coming back from `getDisplayState`.
+Add the `useAserto` hook dependency below the `useAuth` dependency:
+
+```javascript
+import { useAserto } from "@aserto/aserto-react";
+```
+
+Then, find the code block:
+
+```javascript
+const auth = useAuth();
+const isAuthenticated = auth.userData?.id_token ? true : false;
+const [message, setMessage] = useState(false);
+```
+
+And add the following right after it:
+
+```javascript
+const [displayState, setDisplayState] = useState({
```

Comment:
not sure why you need extra state here? once init() is done, you should be able to call getDisplayState